content/blog/2018-12-25-darktable-2.6/2018-12-25-darktable-2.6.md

```diff
@@ -0,0 +1,1167 @@
+skin while preserving the hair of the beard:
+
+  <div style="text-align: center">
+![Before/after, preserving beard]({filename}rt-beard-preserve.jpg)
+  </div>
+
+(Just one "heal" shape on scale 7, propagated to scale 5 using the
+merge slider)
+
+#### Example 2: playing with the skin's texture
+
+If we want to change the texture of the skin on the cheek, we can
+apply a blur on a shape like this:
+
+  <div style="text-align: center">
+![Blur on a detail scale]({filename}rt-blur-detail.jpg)
+  </div>
+
+and get the following before/after result:
+
+  <div style="text-align: center">
+![Changing the skin texture]({filename}rt-skin-texture.jpg)
+  </div>
+
+Obviously, this kind of retouching should be done with great care:
+when pushed too far, one gets an overly artificial look. When unsure,
+you can always get back to your retouch and use blending with an
+opacity lower than 100%, or change the opacity or blur radius of each
+shape individually.
+
+#### Example 3: having fun with the residual image
+
+Just for fun (do not reproduce at home, ugly images to be expected!),
+we can get a tattoo effect by using the clone tool on the residual
+image:
+
+  <div style="text-align: center">
+![Cloning on the residual image]({filename}rt-clone-residual-image.jpg)
+  </div>
+
+While not really elegant, this example illustrates the
+"split-frequency" principle: we've kept the fine details from the
+cheek, and cloned the coarse ones in the residual image.
+
+<span id="filmic" />
+## New module: *filmic*
+
+The filmic module was designed to reproduce the good properties of
+analog film, while giving you the easy controls of digital
+photography. It can be used on any image as a replacement for the
+*base curve* module, and is particularly adapted for images with a
+high dynamic range, i.e. a large difference between brightest and
+darkest areas.
+
+Let's take an example of such image:
+
+  <div style="text-align: center">
+  ![Example HDR image]({filename}filmic-example-image.jpg)
+  </div>
+
+A common approach to deal with high dynamic range images is to
+compress the global contrast while retaining the local one. darktable
+has several modules able to do this: *tone mapping*, *global tonemap*,
+*shadows and highlights*, and since darktable 2.2 the exposure fusion
+mode in the *base curve* module. This contrast compression works to
+some extend, but results in artificial look when pushed too far. What
+you typically want to avoid is this (using *tone mapping*, contrast
+compression set to the maximum):
+
+  <div style="text-align: center">
+  ![Example HDR image]({filename}filmic-hdr-too-much.jpg)
+  </div>
+
+The filmic module shows that another approach is possible, and usually
+gives more natural results. It considers pixels individually, without
+trying to distinguish between local and global contrast. If filmic
+compresses the contrast too much, it is still possible to recover
+local contrast with the excellent *local contrast* module for example.
```

Comment:
> local contrast is applied before filmic and will not recover the compression filmic does in highlights.

Technically, it pre-amplifies the local contrast so that the contrast compression yields a reasonable contrast at the end, but from the user point of view, it doesn't change much. In both cases, the result is augmented local contrast.

> The only reliable way to get the highlights uncompressed is to used a parametric mask with a key blending on luminance, and some feathering and blurring on the mask to avoid halos.

The sentence you're quoting talks about recovering the *local* contrast. A mask will not only recover the local contrast. On this particular image, a mask is actually hard to get right: the dog's hair get caught by the parametric mask, and most combinations of feathering/bluring result in ugly halos. One advantage of "local contrast" is that it can be used with a very approximate mask.